enh-ruby-mode.el

```diff
@@ -741,20 +741,21 @@ Warning: does not play well with command ‘electric-indent-mode’."
 The block must begin on the current line or above it and end after the point.
 If the result is do-end block, it will always be multiline."
   (interactive)
-  (let ((start (point)) beg end)
-    (end-of-line)
-    (unless
-        (if (and (re-search-backward "\\(?:[^#]\\)\\({\\)\\|\\(\\_<do\\_>\\)")
-                 (progn
-                   (goto-char (or (match-beginning 1) (match-beginning 2)))
-                   (setq beg (point))
-                   (save-match-data (enh-ruby-forward-sexp))
-                   (setq end (point))
-                   (> end start)))
-            (if (match-beginning 1)
-                (enh-ruby-brace-to-do-end beg end)
-              (enh-ruby-do-end-to-brace beg end)))
-      (goto-char start))))
+  (let* ((pos (point))
+         (block-start (save-excursion
+                        (end-of-line)
+                        (while (and (not (bobp))
+                                    (not (enh-ruby-point-block-p)))
+                          (backward-char))
```

Comment:
Not sure if this is the most performant way to search backward for a block using syntax properties. This function would _probably_ be slow in 0.1% of the cases when dealing with giant blocks or giant files. But it would work very well for the majority of use cases.